src/body/incoming.rs

```
@@ -270,7 +270,16 @@ impl Body for Incoming {
                         ping.record_non_data();
                         Poll::Ready(Ok(t.map(Frame::trailers)).transpose())
                     }
-                    Err(e) => Poll::Ready(Some(Err(crate::Error::new_h2(e)))),
+                    Err(e) => {
+                        match e.reason() {
+                            // These reasons should cause reading the trailers to stop, but not fail it.
+                            // The same logic as for `Read for H2Upgraded` is applied here.
```

I realize this is just bring down the logic that already exists for data frames, and the code itself looks fine. And the unit tests make sense why they want to suppress the error.

But, reading it all again, I have a worry: couldn't this also be swallowing _real_ cancellations, and the user thinks they got a full body, when they got a truncated one?